### PR TITLE
rgw: fix some list buckets handle leak

### PR DESCRIPTION
**Pull request by tianshan**
Fixes: https://tracker.ceph.com/issues/44283

Signed-off-by: Tianshan Qu <tianshan@xsky.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`

</details>

### CODE REVIEW

**Comment by tianshan** [comment]
@yuvalif fixed, and also include the formatter's cleanup

**Comment by cbodley** [comment]
jenkins test make check

**Comment by cbodley** [comment]
jenkins test make check

File: src/rgw/rgw_bucket.cc [hunk 1]
```diff
@@ -2051,6 +2052,11 @@ static int process_stale_instances(RGWRados *store, RGWBucketAdminOpState& op_st
   bool truncated;
 
   formatter->open_array_section("keys");
+  auto g = make_scope_guard([&store, &handle, &formatter]() {
+                              store->meta_mgr->list_keys_complete(handle);
```
Comment (cbodley) [inline src/rgw/rgw_bucket.cc]: could you please rebase? this part doesn't compile with:
```
src/rgw/rgw_bucket.cc:1604:12: error: 'class rgw::sal::RGWRadosStore' has no member named 'meta_mgr'
     store->meta_mgr->list_keys_complete(handle);
            ^~~~~~~~
```

**Comment by yuriw** [comment]
[wip-yuri8-testing-2019-12-11-2105](https://trello.com/c/QPjkDkCb/778-wip-yuri8-testing-2019-12-11-2105)

**Comment by cbodley** [comment]
@tianshan please rebase

**Comment by tianshan** [comment]
@cbodley rebase done. test failure not related

**Comment by cbodley** [comment]
thanks @tianshan - it looks like it is actually failing to compile:
```
src/rgw/rgw_bucket.cc: In static member function 'static int RGWBucketAdminOp::info(rgw::sal::RGWRadosStore*, RGWBucketAdminOpState&, RGWFormatterFlusher&)':
src/rgw/rgw_bucket.cc:1606:12: error: 'class rgw::sal::RGWRadosStore' has no member named 'meta_mgr'
 1606 |     store->meta_mgr->list_keys_complete(handle);
      |            ^~~~~~~~
```

**Comment by tianshan** [comment]
@cbodley  sorry my mistake, it should be fixed now.

**Comment by ivancich** [comment]
@tianshan Would you add a tracker for this?

**Comment by tianshan** [comment]
@ivancich updated

**Comment by ivancich** [comment]
> @ivancich updated

Thanks, @tianshan . I needed to make some updates to the tracker. Please take a look at the changes I needed made for next time. 

But I expect this will be picked up for a QA run in the near future. Thank you for your help!

**Comment by ivancich** [comment]
QA Results:
http://pulpito.ceph.com/ivancich-2020-05-08_18:24:50-rgw-wip-eric-testing-distro-basic-smithi/
http://pulpito.ceph.com/ivancich-2020-05-08_18:25:10-rgw-wip-eric-testing-distro-basic-smithi/

Any failures  are from other sources and do not represent regressions.

**Pull request merged**